model-optimizer/extensions/back/names_uniqueness_check.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from extensions.back.pass_separator import BackFinish
+from mo.back.replacement import BackReplacementPattern
+from mo.graph.graph import Graph, rename_node
+
+
+class NamesUniquenessCheck(BackReplacementPattern):
+    enabled = True
+
+    def run_after(self):
+        return [BackFinish]
```

Comment:
Why you run the transformation after BackFinish ? Why not before ?

Reply:
I think this check should be as late as possible in order to check an IR close to the final form.